test/validation/api/ipsec/ipsec.c

```diff
@@ -533,17 +535,33 @@ static int recv_pkts_inline(const ipsec_test_part *part,
 
 		ev = odp_queue_deq(suite_context.queue);
 		if (ODP_EVENT_INVALID != ev) {
+			odp_packet_t pkt;
+			int num_pkts = 0;
+
 			CU_ASSERT(odp_event_is_valid(ev) == 1);
-			CU_ASSERT_EQUAL(ODP_EVENT_PACKET,
-					odp_event_types(ev, &subtype));
-			CU_ASSERT_EQUAL(ODP_EVENT_PACKET_IPSEC,
-					subtype);
+			CU_ASSERT_EQUAL(ODP_EVENT_PACKET, odp_event_type(ev));
+			pkt = odp_packet_from_event(ev);
+
 			CU_ASSERT(!part->out[i].status.error.sa_lookup);
 
-			pkto[i] = odp_ipsec_packet_from_event(ev);
-			CU_ASSERT(odp_packet_subtype(pkto[i]) ==
-				  ODP_EVENT_PACKET_IPSEC);
-			i++;
+			if (suite_context.reass_ipv4 || suite_context.reass_ipv6) {
+				reass_status = odp_packet_reass_status(pkt);
+				if (ODP_PACKET_REASS_INCOMPLETE != reass_status) {
+					pkto[i] = pkt;
+					num_pkts = 1;
+				} else {
+					odp_packet_reass_partial_state(pkt, &pkto[i], &reass_state);
```

Comment:
This is slightly fragile as pkto does not have enough room for the maximum number of fragments if i is greater than 0. I am not sure what the best fix would be. Maybe to use temporary frags[MAX_FRAGS] buffer for the call and copying only at most MAX_FRAGS - i frags in pkto (or maybe just the first one, as that would be enough for now).

Reply:
> Maybe to use temporary frags[MAX_FRAGS] buffer for the call and copying only at most MAX_FRAGS - i frags in pkto

Agreed. Will add this in next version.